main.cpp

```diff
@@ -375,32 +484,65 @@ struct FXSection
 {
     float reverbAmount = 50;
     int distAmount = 1;
-    int wetSignal = 50;
+    int delayWetSignal = 50;
     int delayType = 1; //1 = pingpong 2 = tape
     int reverbTypeA = 1; //1 = plate 2 = spring
+    FXSection();
 
-    void delayActive(bool delayOn = false);     
-    float reverbOutput(int reverbType, bool reverbOn = true); //returns one sample of computed reverb
+    void delayActive(bool delayOn, int wetSignalChange);     
+    int delayTypeOut(int delayType); 
     void saturationActive(int distAmount2, bool active);
 };
-void FXSection::delayActive(bool delayOn)
+
+FXSection::FXSection()
 {
-    if (delayOn == true)
+    reverbAmount = 50;
+    distAmount = 1;
+    delayWetSignal = 50;
+    delayType = 1; //1 = pingpong 2 = tape
+    reverbTypeA = 1; //1 = plate 2 = spring
+}
+int FXSection::delayTypeOut(int delayTypeSelector) 
+{
+    if (delayTypeSelector == 1)
     {
-        wetSignal = 75;
+        std::cout << "PingPong Delay \n";
         delayType = 1;
     }
+    else
+    {
+        std::cout << "Tape Delay \n";
+        delayType = 2;
+    }
+    return delayType;
 }
-
-float FXSection::reverbOutput(int reverbType, bool reverbOn)
+void FXSection::delayActive(bool delayOn, int wetSignalChange)
 {
-    if (reverbOn == true)
+    if (delayOn == true)
     {
-        return reverbType * reverbAmount;
+        delayWetSignal += wetSignalChange;
+        std::cout << "delay type is " << delayType <<std::endl;//i want to nest the previous function here so it says Ping Pong or Tape Delay here instead of the int.
+        std::cout << "WetSignal = " << delayWetSignal << std::endl;
     }
-    return 0.0f;
+    else
+        std::cout << "Delay inactive.\n";
 }
 
+/*int FXSection::delayTypeOut(int delayTypeSelector) 
+{
+    if (delayTypeSelector == 1)
+    {
+        delayType = 1;
+        std::cout << "PingPong Delay \n";
+    }
+    else
+    {
+        delayType = 2;
+        std::cout << "Tape Delay \n";
+    }
+    return delayType;
+}
+*/
```

Comment:
Why is this commented out?  it's perfectly good code...

Reply:
I had copied above.  So I commented it out so it wouldn't executed again.  Forgot to remove it.